src/sandstorm/gateway.c++

```diff
@@ -174,6 +181,15 @@ bool isAllowedBasicAuthUserAgent(kj::StringPtr ua) {
 }
 
 kj::Promise<void> GatewayService::request(
+    kj::HttpMethod method, kj::StringPtr url, const kj::HttpHeaders& headers,
+    kj::AsyncInputStream& requestBody, Response& origResponse) {
+
+  auto response = kj::heap<util::http::ExtraHeadersResponse>(origResponse, defaultHeaders.cloneShallow());
```

Comment:
If `ExtraHeadersResponse` kept `defaultHeaders` as a const reference, then the clone wouldn't be needed here. It would be safe since `GatewayService` definitely outlives the request/response.